src/Kiota.Builder/Refiners/TypeScriptRefiner.cs

```diff
@@ -106,6 +106,7 @@ public override void Refine(CodeNamespace generatedCode)
             if(currentMethod.IsOfKind(CodeMethodKind.RequestExecutor))
                 currentMethod.Parameters.Where(x => x.IsOfKind(CodeParameterKind.ResponseHandler) && x.Type.Name.StartsWith("i", StringComparison.OrdinalIgnoreCase)).ToList().ForEach(x => x.Type.Name = x.Type.Name[1..]);
             currentMethod.Parameters.Where(x => x.IsOfKind(CodeParameterKind.Options)).ToList().ForEach(x => x.Type.Name = "RequestOption[]");
+            currentMethod.Parameters.Where(x => x.IsOfKind(CodeParameterKind.Options)).ToList().ForEach(x => { x.Type.Name = "RequestOption[]"; x.Type.ActionOf = false; }); ;
```

Comment:
@baywet what is the purpose of ActionOf?

Reply:
signifies it should be a callback.
ActionOf:true `x: (y: myType) => void`
ActionOf:false `x: myType`

Reply:
Is that applicable to Headers?

https://github.com/microsoft/kiota/blob/48cdebcf9ef5ac0969c3fb73510742699594f9c2/src/Kiota.Builder/KiotaBuilder.cs#L656

Reply:
in dotnet yes because of the semantics, in TypeScript since we have decided on the inline object/record syntax I don't think it provides value, so it should be switched to false by the refiner.
@andrueastman what would you think about using objects for those parameters like TS? I've been switching back and forth between languages too much to have a strong opinion anymore.
So instead of
```CSharp
client.Me.GetAsync(x => { x.Select = "id"; x.Expand = "Manager"});
```
We'd have
```CSharp
client.Me.GetAsync(new() { Select = "id", Expand = "Manager"});
```
Which supposes the consumers IDE/compiler supports C# 10, otherwise they'd have to do
```CSharp
client.Me.GetAsync(new GetQueryParameters { Select = "id", Expand = "Manager"});
```
where GetQueryParameters might conflict with other class names if they are making multiple calls for different endpoints.

Reply:
If `ActionOf` is applicable only to dotnet or a minority of languages then we can default it to false and let the refiner set it to true.

Reply:
maybe but I'd defer that work to @andrueastman in a separate issue. Please log an issue if you feel like this is something we should change.